src/status_im/data_store/realm/core.cljs

```diff
@@ -131,7 +131,9 @@
   ([realm schema-name obj]
    (create realm schema-name obj false))
   ([realm schema-name obj update?]
-   (.create realm (name schema-name) (clj->js obj) update?)))
+   (let [obj-to-save (select-keys obj (keys (get-in entity->schemas
+                                                    [schema-name :properties])))]
+     (.create realm (name schema-name) (clj->js obj-to-save) update?))))
```

Comment:
Given those schemas are known at compilation time we might push it further and pre-create a bunch of optimized serializer/deserializer via macros that would only use relevant data and manipulate only the necessary JS.

That would completely remove the need to dynamically filter the clojure map. It would also remove the use of `clj->js`, notoriously slow because it has to guess the type of each data structure recursively.

Reply:
Yes, that's good idea and I'm very much of favor of such approach !

Reply:
See https://github.com/status-im/status-react/issues/4106